### PR TITLE
handle nan and inf float values

### PR DESCRIPTION
**Pull request by dberenbaum**
Partial fix for https://github.com/iterative/dvcx/issues/1697#issuecomment-2269457257:

- [x] make sure NaN and Inf are supported in SQLite
- [ ] NaN and Inf supported in CH - see https://github.com/iterative/studio/pull/10429
- [x] from_parquet() support NaN and Inf
- [ ] make sure `Array(Float)` support that - needs follow-up since orjson does not support [nan/inf](https://github.com/ijl/orjson/issues/488)

This PR treats all missing values in float columns as NaN for a couple reasons:
1. Consistency with the pydata ecosystem. Numpy and [pandas](https://pandas.pydata.org/docs/user_guide/missing_data.html) both treat all missing values in float columns as NaN.
2. Sqlite does not differentiate between null and NaN (see [sqlite](https://stackoverflow.com/questions/67360775/sql-command-to-insert-inf-nan-and-inf-into-sqlite3-database) [discussions](https://stackoverflow.com/questions/20619957/nan-in-sqlite3-have-invalid-numerical-comparisons) and [source code](https://github.com/sqlite/sqlite/blob/210f6f89befbe4e4c561fc34aabc175f552450f4/src/vdbemem.c#L980)), although [CH does](https://clickhouse.com/docs/en/sql-reference/data-types/float#nan-and-inf).

### CODE REVIEW

File: src/datachain/lib/dc.py [hunk 1]
```diff
@@ -1414,7 +1415,14 @@ def from_csv(
 
         parse_options = ParseOptions(delimiter=delimiter)
         read_options = ReadOptions(column_names=column_names)
-        format = CsvFileFormat(parse_options=parse_options, read_options=read_options)
+        convert_options = ConvertOptions(
+            strings_can_be_null=True, null_values=STR_NA_VALUES
+        )
```
Comment (dberenbaum) [inline src/datachain/lib/dc.py:1420]: This provides pandas-like null handling for csv files

File: tests/unit/lib/test_datachain_merge.py [hunk 2]
```diff
@@ -45,12 +47,12 @@ class TeamMember(BaseModel):
 
 
 def test_merge_objects(test_session):
+    skip_if_not_sqlite()
```
Comment (dberenbaum) [inline tests/unit/lib/test_datachain_merge.py:50]: See the note about joins in https://github.com/iterative/studio/pull/10429

File: tests/unit/lib/test_datachain_merge.py [hunk 3]
```diff
@@ -103,14 +105,13 @@ def test_merge_similar_objects(test_session):
 
 
 def test_merge_values(test_session):
+    skip_if_not_sqlite()
```
Comment (dberenbaum) [inline tests/unit/lib/test_datachain_merge.py:108]: See the note about joins in https://github.com/iterative/studio/pull/10429

**Comment by dberenbaum** [comment]
Remaining test failures are either unrelated or require https://github.com/iterative/studio/pull/10429

File: tests/unit/lib/test_datachain.py [hunk 4]
```diff
@@ -1497,3 +1506,48 @@ def test_mutate_with_expression_without_type(catalog):
     assert str(excinfo.value) == (
         "Error for column new: Cannot infer type with expression id - :id_1"
     )
+
+
+def test_from_values_nan_inf(tmp_dir, catalog):
+    vals = [float("nan"), float("inf"), float("-inf")]
```
Comment (dmpetrov) [inline tests/unit/lib/test_datachain.py:1512]: that's impressive!

**Review by dmpetrov — APPROVED** [review]
Really cool 🪄 
I didn't expect this can support this as naturally.

**Comment by dberenbaum** [comment]
Going to merge but we should follow up on arrays and CH joins

**Comment by shcheklein** [comment]
> Going to merge but we should follow up on arrays and CH joins

@dberenbaum can we please review and create a ticket in the DataChain repo for this if needed?

**Comment by dberenbaum** [comment]
> * make sure `Array(Float)` support that - needs follow-up since orjson does not support [nan/inf](https://github.com/ijl/orjson/issues/488)

Opened https://github.com/iterative/datachain/issues/386. CH was fixed by https://github.com/iterative/studio/pull/10471.

**Pull request merged**